xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -977,6 +979,35 @@ private static String serializePropertyValue(PropertyInterface property)
         }
     }
 
+    /**
+     * Serializes the value of the given XObject property. In case the property is an instance of
+     * {@link ComputedFieldClass}, the serialized value is the computed property value.
+     * @param property an XObject property
+     * @param propertyClass the PropertyClass of that XObject proprety
+     * @param context the XWikiContext
+     * @return the String representation of the property value
+     */
+    private static String serializePropertyValue(PropertyInterface property,
+        com.xpn.xwiki.objects.classes.PropertyClass propertyClass, XWikiContext context)
+    {
+        if (propertyClass instanceof ComputedFieldClass) {
+            // TODO: the XWikiDocument needs to be explicitely set in the context, otherwise method
+            //  PropertyClass.renderInContext fires a null pointer exception: bug?
+            XWikiDocument document = context.getDoc();
+            try {
+                context.setDoc(property.getObject().getOwnerDocument());
+                ComputedFieldClass computedFieldClass = (ComputedFieldClass) propertyClass;
+                return computedFieldClass.getComputedValue(propertyClass.getName(), "", property.getObject(), context);
```

Comment:
So if I'm right you can now catch an exception here in case `getComputedValue` didn't worked for some reason. And in the catch I guess you can log an error with the actual exception, and fallback on `serializePropertyValue(property);`.

Reply:
Right, thank you, good idea, I added that. I had to turn the method `ModelFactory:serializePropertyValue(PropertyInterface,PropertyClass,XWikiContext)` to not static in order to access the logger field.